Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealitySearchUtility.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor.Search
+{
+    /// <summary>
+    /// Utility for searching profile 
+    /// </summary>
+    public static partial class MixedRealitySearchUtility
+    {
+        /// <summary>
+        /// Field names that shouldn't be displayed in a profile field search.
+        /// </summary>
+        private static readonly HashSet<string> serializedPropertiesToIgnore = new HashSet<string>()
+        {
+            // Unity base class fields
+            "m_Name",
+            "m_Script",
+            "m_Enabled",
+            "m_GameObject",
+            "m_ObjectHideFlags",
+            "m_CorrespondingSourceObject",
+            "m_PrefabInstance",
+            "m_PrefabAsset",
+            "m_EditorHideFlags",
+            "m_EditorClassIdentifier",
+            // Profile base class fields
+            "isCustomProfile",
+        };
+
+        /// <summary>
+        /// Field types that don't need their child properties displayed.
+        /// </summary>
+        private static readonly HashSet<string> serializedPropertyTypesToFlatten = new HashSet<string>()
+        {
+            typeof(SystemType).Name,
+        };
+
+        private const int minSearchStringLength = 3;
+
+        /// <summary>
+        /// Searches all profiles and sub profiles for matching fields. Returns results in unranked order.
+        /// </summary>
+        /// <param name="profile">The configuration profile you're searching.</param>
+        /// <param name="config">Configuration for search.</param>
+        public static IEnumerable<ProfileSearchResult> SearchProfileFields(UnityEngine.Object profile, SearchConfig config)
```

Comment:
This function is assuming it's going to analyze a MRTK profile. Make the profile argument of type BaseMixedRealityProfile to be explicit